### PR TITLE
Remove old atmosphere rendering

### PR DESCRIPTION
**Pull request by jherico**
The old ugly and broken atmosphere is removed. Content creators should use skyboxes and/or procedural skyboxes to create atmosphere effects instead.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20fac667468254f3c0329b78b788cf6d3d6a48a1 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 60a0869fe866ad09284773e405988910e7500238 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 826ab45aa053694cda8103c7739e0e7d2827b0f6 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea4f52f9daaeb9add4eb46dfff1f75ecf64c10eb available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by ZappoMan** [comment]
Needs to be updated to remove the atmosphere features from the edit.js Entity Properties panel.

**Comment by hifi-gustavo** [comment]
Build console output for commit: eb10dfc76ca7edc329faf4b2e4910a797b816b23 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by ZappoMan** [comment]
We need to make sure to bump `versionForPacketType()` value for `Entity*` packet types in `PacketHeaders.cpp` before we can merge this.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a0a41b4fa333f9fa1db9450403c79cba798190ad available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c078b51918e25c44cc52b0ce7d495ba83b1b28f available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0546537b44b8139bf12b5467d9eeded1ba9c37ac available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae54d3b1cdb6265850b4d7885e1ebaa5ab40fe7a available at: https://gist.github.com/68a042a582d38de663a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 562a038a35e6c3a80b732fc950175f58c74d58e8 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 562a038a35e6c3a80b732fc950175f58c74d58e8 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5379500a82f0e41c284c529ea9f8b69cc11ec427 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5379500a82f0e41c284c529ea9f8b69cc11ec427 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4e5498c5675aaac44d43bc4dfda017a6b36ce1de available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6849.dmg

File: libraries/shared/src/BackgroundMode.h [hunk 1]
```diff
@@ -13,8 +13,9 @@
 
 enum BackgroundMode {
     BACKGROUND_MODE_INHERIT,
-    BACKGROUND_MODE_ATMOSPHERE,
     BACKGROUND_MODE_SKYBOX,
+
+    BACKGROUND_MODE_ITEM_COUNT,
```
Comment (AndrewMeadows) [inline libraries/shared/src/BackgroundMode.h]: This name threw me: "item count"?  I had to track in the code to see how it was being used... "Oh he means BACKGROUND_MODE_COUNT."

**Comment by ZappoMan** [comment]
*Testing Notes - Please test these things before approving:*
- Test cross connecting between old/new servers/clients - you shouldn't crash, you should get "protocol" warning.
- Verify that old atmosphere properties are not longer visible in the edit dialog box
- Verify that skyboxes, procedural skyboxes, still work.
- Verify you can enter/leave zones with different skybox settings and things still work as expected.
- Verify that stars render when no skybox is set.

**Comment by AndrewMeadows** [comment]
There must be a plan for new and improved atmospheric effects.  I idly wonder what it might be although I can be content to know that someone has a plan.

**Comment by ZappoMan** [comment]
@AndrewMeadows - the decision was made to remove this completely, and not have any atmosphere built in. Instead, for MVP, content creators are advised to use skyboxes or procedural shaders.

**Comment by thoys** [comment]
I have test this on Windows, based on @ZappoMan 's Testing Notes:

- Test cross connecting between old/new servers/clients - you shouldn't crash, you should get "protocol" warning.
  - :heavy_check_mark: Check! Received protocol warning, upon arrival in an old server.
- Verify that old atmosphere properties are not longer visible in the edit dialog box
  - :heavy_check_mark: Check! You are no longer able to select atmosphere as background mode, yet there is the stage sunmodel, which is probably there for light purposes.
  - :question: Also, I do see skybox color, is that being used?
- Verify that skyboxes, procedural skyboxes, still work.
  - :heavy_check_mark: Skybox Check!
  - :heavy_check_mark: Procedural Skybox Check!
- Verify you can enter/leave zones with different skybox settings and things still work as expected.
  - :heavy_check_mark: Entering Check! Seeing the skybox as I enter.
  - :heavy_check_mark: Leaving Check! Seeing the stars as I leave.
- Verify that stars render when no skybox is set.
  - :heavy_check_mark: Check!

Congrats @jherico , I have completed the QA test successfully! :bowtie:

**Comment by ZappoMan** [comment]
@thoys - re: " Also, I do see skybox color, is that being used?" -- this is correct. Skybox color is used to render a solid color if no "image" skybox is provided. -- so that result is considered passing. Thanks!

**Comment by thoys** [comment]
Ah @ZappoMan , can you confirm if the Skybox color option is broken in the latest release build? I didn't get it to work for almost a year. It simply shows stars for me when no Image is provided.

**Comment by jherico** [comment]
Adding DNM until the issue servers are updatable, since this bumps the protocol version.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47a1c164309698207297c42b271113cfc3d258b4 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6849.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47a1c164309698207297c42b271113cfc3d258b4 available at: https://gist.github.com/e00d7e5e8b2632dcda18<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6849.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6849.dmg

**Pull request merged**